gqlalchemy/transformations/translators/dgl_translator.py

```diff
@@ -0,0 +1,137 @@
+                # Copy edge features
+                edge_properties = dict(map(lambda pair: (pair[0], to_cypher_value(pair[1][eid])), etype_properties.items()))
+                edge_properties[DGL_ID] = int(eid)
+
+                # Create query
+                queries.append(
+                    self.create_insert_query(
+                        source_node_label,
+                        source_node_properties,
+                        edge_type,
+                        edge_properties,
+                        dest_node_label,
+                        dest_node_properties,
+                    )
+                )
+        return queries
+
+    def get_instance(self):
+        """Create instance of DGL graph from all edges that are inside Memgraph. Currently, isolated nodes are ignored because they don't contribute in message passing neural networks. Only numerical features
+        that are set on all nodes or all edges are transferred to the DGL instance since this is DGL's requirement. That means thay any string values properties won't be transferred, as well as numerical properties
+        that aren't set on all nodes. However, features of type list are transferred to the DGL and can be used as any other feature in the DGL graph. Regardless of data residing inside Memgraph database, the created
+        DGL graph is a heterograph instance.
+        Returns:
+            DGL heterograph instance.
+        """
+        # Get all nodes and edges from the database
+        query_results = self.get_all_edges_from_db() 
+
+        # Parse it into nice data structures
+        src_nodes, dest_nodes, node_features, edge_features, _ = self._parse_mem_graph(query_results)
+
+        graph_data = {}
+        # Iterating over src_nodes and dest_nodes should be the same
+        for type_triplet in src_nodes.keys():
+            graph_data[type_triplet] = (
+                torch.tensor(src_nodes[type_triplet], dtype=torch.int32),
+                torch.tensor(dest_nodes[type_triplet], dtype=torch.int32),
+            )
+
+        # Fail safely when graph is empty
+        if len(graph_data) == 0:
+            return None
+
+        # Create heterograph
+        graph = dgl.heterograph(graph_data)
+        # Set node features
+        for node_label, features_dict in node_features.items():
+            for feature_name, features in features_dict.items():
+                features = Translator.validate_features(features, graph.num_nodes(node_label))
+                if features is not None:
+                    graph.nodes[node_label].data[feature_name] = features
+
+        # Set edge features
+        for edge_triplet, features_dict in edge_features.items():
+            for feature_name, features in features_dict.items():
+                features = Translator.validate_features(features, graph.num_edges(edge_triplet))
+                if features is not None:
+                    graph.edges[edge_triplet].data[feature_name] = features
+
```

Comment:
`
for edge_triplet, features_dict ...
for feature_name, features ...
translated_features = ... //it makes sense here to rename variable
if features is None: //this way you are not indenting lines so much
    continue
graph.edges[edge_triplet].data[feature_name] = translated_features`

Reply:
Agreed